### PR TITLE
Put/delete methods

### PR DESCRIPTION
**Pull request by dlvsn**

### CODE REVIEW

**Comment by dlvsn** [comment]
I don't know how to fix problems with ci

**Comment by dlvsn** [comment]
everything works on my computer

File: src/main/java/org/example/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -30,9 +38,17 @@ public BookDto getBookById(@PathVariable Long id) {
...
-        return bookService.save(bookDto);
+    @PutMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public BookDto updateBook(@PathVariable Long id,
+                              @RequestBody CreateRequestBookDto createRequestBookDto) {
+        return bookService.update(id, createRequestBookDto);
     }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.ACCEPTED)
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/onlinebookstore/controller/BookController.java:49]:
```suggestion
    @ResponseStatus(HttpStatus.NO_CONTENT)
```

File: src/main/java/org/example/onlinebookstore/service/BookServiceImpl.java [hunk 2]
```diff
@@ -35,4 +35,16 @@ public BookDto findById(Long id) {
                 new EntityNotFoundException("Can't find book by id: " + id));
         return bookMapper.toDto(bookFromDb);
     }
+
+    @Override
+    public BookDto update(Long id, CreateRequestBookDto bookDto) {
+        Book model = bookMapper.toModel(bookDto);
+        model.setId(id);
+        return bookMapper.toDto(bookRepository.save(model));
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/onlinebookstore/service/BookServiceImpl.java:43]:
```suggestion
        bookMapper.updateBookFromDto(bookDto, book);
        return bookMapper.toDto(bookRepository.save(model));
```

File: src/main/java/org/example/onlinebookstore/service/BookServiceImpl.java [hunk 3]
```diff
@@ -35,4 +35,16 @@ public BookDto findById(Long id) {
                 new EntityNotFoundException("Can't find book by id: " + id));
         return bookMapper.toDto(bookFromDb);
     }
+
+    @Override
+    public BookDto update(Long id, CreateRequestBookDto bookDto) {
+        Book model = bookMapper.toModel(bookDto);
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/onlinebookstore/service/BookServiceImpl.java:41]:
```suggestion
        Book book = bookMapper.toModel(bookDto);
```

File: pom.xml [hunk 4]
```diff
@@ -64,10 +64,12 @@
...
             <scope>test</scope>
         </dependency>
+
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
+
         <dependency>
             <groupId>org.projectlombok</groupId>
```
Comment (Elena-Bruyako) [inline pom.xml:74]: you have duplicate lombok dependency, remove one of them

File: pom.xml [hunk 5]
```diff
@@ -114,7 +128,22 @@
...
+                    <propertyFile>src/main/resources/liquibase.properties</propertyFile>
+                </configuration>
+                <executions>
+                    <execution>
+                        <phase>process-resources</phase>
+                        <goals>
+                            <goal>update</goal>
+                        </goals>
+                    </execution>
+                </executions>
```
Comment (Elena-Bruyako) [inline pom.xml:145]: remove
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/org/example/onlinebookstore/controller/BookController.java [hunk 6]
```diff
@@ -30,9 +38,17 @@ public BookDto getBookById(@PathVariable Long id) {
         return bookService.findById(id);
     }
 
-    @PostMapping
-    @ResponseStatus(HttpStatus.CREATED)
-    public BookDto createBook(@RequestBody CreateRequestBookDto bookDto) {
-        return bookService.save(bookDto);
+    @PutMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
```
Comment (liliia-ponomarenko) [inline src/main/java/org/example/onlinebookstore/controller/BookController.java:42]: it's default status code, so you can remove this annotation

File: src/main/java/org/example/onlinebookstore/controller/BookController.java [hunk 7]
```diff
@@ -30,9 +38,17 @@ public BookDto getBookById(@PathVariable Long id) {
...
+                              @RequestBody CreateRequestBookDto createRequestBookDto) {
+        return bookService.update(id, createRequestBookDto);
     }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteBook(@PathVariable Long id) {
+        bookService.delete(id);
+    }
+
```
Comment (liliia-ponomarenko) [inline src/main/java/org/example/onlinebookstore/controller/BookController.java:53]:
```suggestion
```

File: src/main/resources/db/changelog/changes/01-create-book-table.yaml [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/01-create-book-table.yaml:48]: add one empty line to the end of the file ;)

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! A few minor comments;)

**Pull request merged**